### PR TITLE
[MTM-56083]: user guide for device's JWT token using X.509 certificates

### PR DESCRIPTION
**Pull request by cumulocitypans**
Documentation is available on :
https://sagportal.sharepoint.com/:w:/r/sites/IoTAnalyticsRnDOps/_layouts/15/doc2.aspx?sourcedoc=%7B489B6AA8-3AFA-44DC-BD83-32DDC4B93F43%7D&file=Enable%20mTLS%20for%20REST.docx&action=default&mobileredirect=true

https://sagportal.sharepoint.com/:w:/r/sites/IoTAnalyticsRnDOps/_layouts/15/Doc.aspx?sourcedoc=%7B6007259B-A7E5-46C4-9E13-F34EAAF6CAFA%7D&file=Tech_DOC_Enable_MTLS_For_Rest_API.docx&action=default&mobileredirect=true

### CODE REVIEW

File: content/device-integration/device-rest-client-bundle/rest-client-java-certificates.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+private static final String TRUSTSTORE_FORMAT = "jks";
+
+private static final String LOCAL_DEVICE_CHAIN = "-----BEGIN CERTIFICATE----- MIIcQhNJJ0F/lfjm -----END CERTIFICATE-----";
+private static final String PLATFORM_URL = "<URL of the platform>";
+```  
+
+The device can now generate JWT token. Note that before the first connect no other actions are required, for example, creating a user. The user is created during the [auto registration](/device-integration/certificate/#device-certificates) process.
+
+{{< c8y-admon-info >}}
+You do not need to set a password, user or tenant for the REST java client using certificates. {{< product-c8y-iot >}} will recognize the tenant and the user by the provided certificate.
```
Comment (Michal-Bak) [inline content/device-integration/device-rest-client-bundle/rest-client-java-certificates.md:72]: It is a duplication? 
I think we have already description:
 how to generate token(To generate a valid certificate)  
how to upload (To copy and upload the certificate)
To change the configuration -> probably need to be explained with the same step where you describe how Create a Java application.

**Review by Michal-Bak — CHANGES_REQUESTED** [review]

**Comment by Michal-Bak** [comment]
And I believe you do not need to fork the repository, Since you were added toSoftwareAG organization I guess you can open PR directly

File: content/device-integration/certificate-bundle/device-certificates.md [hunk 2]
```diff
@@ -30,7 +19,7 @@ Devices connecting to the platform with certificates do not need to provide the
 
 **Auto registration**
 
-The user for the device will be created during the first MQTT call, if a device certificate is derived from a trusted certificate which was uploaded to the {{< product-c8y-iot >}} platform with a flag _autoRegistrationEnabled_ with a value of true.
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/device-certificates.md:33]: Hi @Michal-Bak , I moved this up in certificate-bundle which is common for both REST and MQTT

File: content/device-integration/device-rest-client-bundle/device-access-token.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    curl -v -cert domain-cert.pem -key domain-private-key.pem \
+       -H 'Accept: application/json' \
+       -H 'X-Ssl-Cert-Chain:<cert-chain>' \
+       -X POST \
+       https://<{{< product-c8y-iot >}} tenant domain>/devicecontrol/deviceAccessToken
+
+Replace `<cert-chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}.
+
+The same credentials used to access the {{< product-c8y-iot >}} Web UI can be used to execute the REST calls.
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/device-rest-client-bundle/device-access-token.md:26]: Agree, will change sentence it's not serve purpose.

File: content/device-integration/device-rest-client-bundle/rest-client-java.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+		System.out.println(response.body());
+	}
+
+	/**
+	 * Gets the SSL context.
+	 *
+	 * @param trustManagerFactory the trust manager factory
+	 * @param keyManagerFactory the key manager factory
+	 * @return the SSL context
+	 */
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/device-rest-client-bundle/rest-client-java.md:118]: will remove.

File: content/device-integration/certificate-bundle/device-certificates.md [hunk 5]
```diff
@@ -1,25 +1,14 @@
 ---
-weight: 80
-title: Device certificates
+weight: 70
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/device-certificates.md:2]: weight decide page sequence like lighter weight come first.

File: content/device-integration/certificate-bundle/device-certificates.md [hunk 6]
```diff
@@ -1,25 +1,14 @@
...
+title: Device certificate
 layout: redirect
 ---
 
-### Overview {#overview}
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/device-certificates.md:7]: it is common for Rest and MQTT thats why I removed in one common place in certificate-bundle

File: content/device-integration/certificate-bundle/overview.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+Each tenant individually defines whom it trusts by uploading the base CA certificate.
+
+Devices connecting to the platform with certificates do not need to provide the tenant ID, username and password. Authentication information will be obtained from the certificates.
+
+
+### Prerequisites {#prerequisites}
+
+In order to follow this tutorial, check if the following prerequisites are met:
+
+-   You have a valid tenant, user and password in order to access {{< product-c8y-iot >}}.
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/overview.md:20]: For doing admin related activity like uploading CA certificate in platform. Its perquisite

File: content/device-integration/device-rest-client-bundle/rest-client-java-certificates.md (hunk 1 above)
Comment (cumulocitypans) [reply content/device-integration/device-rest-client-bundle/rest-client-java-certificates.md]: Removed

File: content/device-integration/device-rest-client.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+weight: 40
+title: JWT token using REST client
+layout: bundle
+section:
+  - device_management
+---
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/device-rest-client.md]: I did, only thing pending is creating subsection in section.

File: content/device-integration/certificate-bundle/device-certificates.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/device-certificates.md:33]: This section moved to common place under certificate-bundle folder.

File: content/device-integration/certificate-bundle/device-certificates.md [hunk 9]
```diff
@@ -49,29 +38,6 @@ Single registration is not supported for devices which are going to use certific
 During device registration, the device user is created, which is necessary for device communication with the platform.
 {{< /c8y-admon-info >}}
 
-### JWT token retrieval {#jwt-token-retrieval}
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/certificate-bundle/device-certificates.md:52]: It is moved in MQTT section under mqtt-bundle/mqtt-examples-bundle

**Comment by cumulocitypans** [comment]
As per discussion with @BeateRixen and @sarangovindarajan created a new PR https://github.com/SoftwareAG/c8y-docs/pull/1795 with latest changes. Please review and provide feedback.

**Pull request closed without merging**